core_acc_analysis/stable_gene_relationships.py

```diff
@@ -0,0 +1,397 @@
+pa14_similarity_scores_filename = "../core_core_analysis/pa14_similarity_scores.tsv"
+
+pao1_similarity_scores = pd.read_csv(
+    pao1_similarity_scores_filename, sep="\t", header=0, index_col=0
+)
+pa14_similarity_scores = pd.read_csv(
+    pa14_similarity_scores_filename, sep="\t", header=0, index_col=0
+)
+
+# +
+# Get most and least stable core genes
+pao1_most_stable_genes = list(
+    pao1_similarity_scores[pao1_similarity_scores["label"] == "most stable"].index
+)
+pao1_least_stable_genes = list(
+    pao1_similarity_scores[pao1_similarity_scores["label"] == "least stable"].index
+)
+
+pa14_most_stable_genes = list(
+    pa14_similarity_scores[pa14_similarity_scores["label"] == "most stable"].index
+)
+pa14_least_stable_genes = list(
+    pa14_similarity_scores[pa14_similarity_scores["label"] == "least stable"].index
+)
+# -
+
+# %%time
+expression_dist_counts_pao1_most = (
+    gene_relationships.get_relationship_in_expression_space(
+        pao1_corr,
+        pao1_most_stable_genes,
+        pao1_arr,
+        offset_to_bin,
+        pao1_operon_expression_to_use,
+        sum_increment_to_use,
+    )
+)
+expression_dist_counts_pao1_least = (
+    gene_relationships.get_relationship_in_expression_space(
+        pao1_corr,
+        pao1_least_stable_genes,
+        pao1_arr,
+        offset_to_bin,
+        pao1_operon_expression_to_use,
+        sum_increment_to_use,
+    )
+)
+
+# %%time
+expression_dist_counts_pa14_most = (
+    gene_relationships.get_relationship_in_expression_space(
+        pa14_corr,
+        pa14_most_stable_genes,
+        pa14_arr,
+        offset_to_bin,
+        pa14_operon_expression_to_use,
+        sum_increment_to_use,
+    )
+)
+expression_dist_counts_pa14_least = (
+    gene_relationships.get_relationship_in_expression_space(
+        pa14_corr,
+        pa14_least_stable_genes,
+        pa14_arr,
+        offset_to_bin,
+        pa14_operon_expression_to_use,
+        sum_increment_to_use,
+    )
+)
+
+expression_dist_counts_pao1_most.head()
+
+expression_dist_counts_pao1_least.head()
+
+expression_dist_counts_pa14_most.head()
+
+expression_dist_counts_pa14_least.head()
+
+# ### Plot
+
+# +
+# Plot PAO1 trends
+fig, axes = plt.subplots(ncols=1, nrows=2, figsize=(10, 10))
+
+fig = sns.barplot(
+    data=expression_dist_counts_pao1_most,
+    x="offset",
+    y="total",
+    hue="gene type",
+    ax=axes[0],
+    palette=sns.color_palette("Paired"),
+)
+fig.legend_.remove()
+fig.set_title("Starting with most stable core gene PAO1")
+fig.set_ylabel("Number of genes")
+fig.set_xlabel("Rank correlation in expression space")
+
+fig = sns.barplot(
+    data=expression_dist_counts_pao1_least,
+    x="offset",
+    y="total",
+    hue="gene type",
+    ax=axes[1],
+    palette=sns.color_palette("Paired"),
+)
+fig.legend_.remove()
+fig.set_title("Starting with least stable core gene PAO1")
+fig.set_ylabel("Number of genes")
+fig.set_xlabel("Rank correlation in expression space")
+
```

Comment:
small question -- are you interested in the proportion of genes that are highly correlated, or the absolute number? I'm just wondering is a stacked proportional bar chart would be better here because then you might be able to plot more modules, but not sure how important that is.

Reply:
This is a great point! My next step is to figure out what is the best way to visualize this data because, I agree, I think what would be better is if there is some sort of normalization of the data so that we can see proportionally who is related to who.